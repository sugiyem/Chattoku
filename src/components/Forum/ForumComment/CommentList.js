import { useEffect, useState } from "react";
import { SectionList } from "react-native";
import FetchComment from "./FetchComment";
import CommentCard from "./CommentCard";

const CommentList = ({ forumId, postId }) => {
  const [comments, setComments] = useState([]);

  useEffect(() => {
    FetchComment(
      forumId,
      postId,
      (data) => setComments(data),
      () => {}
    );
  }, []);

  const renderItem = ({ section, item }) => {
    console.log(item);
    return (
      <CommentCard
        {...item}
        forumId={forumId}
        postId={postId}
        setComments={setComments}
      />
    );
  };

  const renderHeader = () => <></>;

  const renderFooter = () => <></>;

  return (
    <>
      <SectionList
        removeClippedSubviews={true}
        sections={[{ data: comments }]}
        renderItem={renderItem}
        renderHeader={renderHeader}
        renderFooter={renderFooter}
      />
    </>
  );
};

export default CommentList;
