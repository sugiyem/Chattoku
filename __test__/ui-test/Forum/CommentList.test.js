import React from "react";
import renderer from "react-test-renderer";
import CommentList from "../../../src/components/Forum/ForumComment/CommentList";

jest.mock("../../../src/services/Forum/FetchComment");
jest.mock("../../../src/services/Forum/HandleForumAdmin");
jest.mock("../../../src/services/Profile/FetchUserInfo");

describe("Test Comment List UI", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(<CommentList forumId="forum-1" postId="post-1" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
