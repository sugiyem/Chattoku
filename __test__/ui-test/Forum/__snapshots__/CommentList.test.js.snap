// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Comment List UI Renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [],
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={true}
  renderFooter={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  style={
    Object {
      "alignSelf": "stretch",
    }
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
